x-pack/plugins/apm/public/utils/url.tsx

```diff
@@ -187,8 +201,9 @@ export const UnconnectedKibanaLink: React.SFC<KibanaLinkArgs> = ({
   const currentQuery = toQuery(location.search);
   const nextQuery = {
     ...query,
-    // use "_g" if it's set in the url, otherwise use default
-    _g: currentQuery._g || DEFAULT_KIBANA_TIME_RANGE,
+    _g: query._g
+      ? rison.encode(query._g)
+      : getCurrentOrDefaultGArg(currentQuery._g),
```

Comment:
Now I see what's going on -- when we removed the ability to override this via `query._g` we removed the ability for the ML link to add in the `ml` key to the `_g` arg

Reply:
Exactly. I didn't think about that back then. Now we have a test for `UnconnectedKibanaLink` (and we might need some more for `ViewMLJob`), so it won't happen again.